byteswriter/writer.go

```diff
@@ -0,0 +1,80 @@
+// Package byteswriter implements a WriteSeeker backed by a
+// dynamically expanding buffer.
+package byteswriter
+
+import (
+	"fmt"
+	"io"
+)
+
+// A Writer implements a WriteSeeker interface backed by a
+// synamically expanding buffer.
+type Writer struct {
+	buf []byte
+	pos int
+}
+
+// New returns a new writer with an initial alloction of 4KB.
+func New() *Writer {
+	ret := new(Writer)
+	ret.buf = make([]byte, 0, 4*1024)
+	return ret
+}
+
+// NewPreallocated returns a new writer with an initial allocation of n.
+func NewPreallocated(n int) *Writer {
+	ret := new(Writer)
+	ret.buf = make([]byte, 0, n)
+	return ret
+}
+
+// Size returns the current size of the buffer.
+func (w *Writer) Size() int {
+	return len(w.buf)
+}
```

Comment:
All the other Size() functions in the io and related packages return int64. Probably best to be consistent with them.